### PR TITLE
Do not use mutable types in QAOA initialization

### PR DESCRIPTION
**Pull request by tbabej**
This pull request (in addition to couple of styling improvements) fixes a potential source of problems:

Default method arguments should never be mutable objects, such as []
or {} - this leads to very hard to debug situations, since these default
values are stored as a part of the function object definition and
subsequently shared between all the instances of the class.

See: http://docs.python-guide.org/en/latest/writing/gotchas/

### CODE REVIEW

File: grove/pyqaoa/qaoa.py [hunk 1]
```diff
@@ -66,68 +66,60 @@ def __init__(self, qvm, qubits, steps=1, init_betas=None,
         :param store_basis: (optional) boolean flag for storing basis states.
                             Default=False.
         """
+
+        # Randomize seed
```
Comment (ampolloreno) [inline grove/pyqaoa/qaoa.py]: I think this comment is a bit misleading - we're setting the seed to the provided seed. From this comment I thought we were picking a random seed. 

Maybe `# Seed the random number generator.` might be better?

File: grove/pyqaoa/qaoa.py [hunk 2]
```diff
@@ -66,68 +66,60 @@ def __init__(self, qvm, qubits, steps=1, init_betas=None,
...
+            }
+        }
+
+        self.betas = init_betas or np.random.uniform(0, np.pi, self.steps)[::-1]
+        self.gammas = init_gammas or np.random.uniform(0, 2*np.pi, self.steps)
+        self.vqe_options = vqe_options or {}
+
+        self.ref_state_prep = (
+            driver_ref or
+            pq.Program(*[H(i) for i in self.qubits])
```
Comment (ampolloreno) [inline grove/pyqaoa/qaoa.py]: I don't think the asterisk is needed here, `Program` can take a list.

**Comment by ampolloreno** [comment]
@tbabej I actually made the two changes on your branch - if they look good to you, we can merge!

**Comment by tbabej** [comment]
Looks good to me! Thanks for merging.

**Pull request merged**